File: ch1/1-14.c
#include <stdio.h>

#define MAX 50

int main(void)
{
    int c;
    int len = 0;
    char chars[MAX] = { 0 };
    int freqs[MAX] = { 0 };

    while((c = getchar()) != EOF) {
        if (c == ' ') {
            continue;
        }

        if (c == '\n') {
            break;
        }

        int in_chars = 0;
        for (int i = 0; i < len; i++) {
            if (chars[i] == c) {
                in_chars = 1;
                freqs[i]++;
                break;
            }
        }

        if (in_chars == 0) {
            chars[len] = c;
            freqs[len] = 1;
            len++;
        }
    }

    for (int i = 0; i < len; i++) {
        printf("%c ", chars[i]);
        for (int j = 0; j < freqs[i]; j++) {
            printf("-");
        }
        printf("\n");
    }
    return 0;
}

File: ch1/1-14.go
package main

import (
	"bufio"
	"fmt"
	"os"
)

const max = 50

func main() {
	var leng int
	chars := make([]rune, max)
	freqs := make([]int, max)
	r := bufio.NewReader(os.Stdin)
	for {
		b, err := r.ReadByte()
		if err != nil {
			break
		}
		c := rune(b)
		if c == ' ' {
			continue
		}
		if c == '\n' {
			break
		}

		var inChars bool
		for i := 0; i < leng; i++ {
			if chars[i] == c {
				inChars = true
				freqs[i]++
				break
			}
		}

		if !inChars {
			chars[leng] = c
			freqs[leng] = 1
			leng++
		}
	}

	for i := 0; i < leng; i++ {
		fmt.Printf("%c", chars[i])
		for j := 0; j < freqs[i]; j++ {
			fmt.Printf("-")
		}
		fmt.Printf("\n")
	}
}
